src/udev/udev-builtin-hwdb.c

```diff
@@ -99,6 +99,18 @@ static int udev_builtin_hwdb_search(sd_device *dev, sd_device *srcdev,
                         last = true;
                 }
 
+                if (streq(dsubsys, "firewire")) {
+                        const char *devname;
+
+                        // The node device has character device while the unit device doesn't.
+                        if (!sd_device_get_devname(d, &devname) && strlen(devname) > 0) {
```

Comment:
Please do not use implicit cast to bool, i.e. please use `sd_device_get_devname(d, &devname) >= 0` instead.
Also, I do not think it is necessary to check `strlen(devname)`. So,
```
if (sd_device_get_devname(d, NULL) >= 0)
```
is sufficient.

Reply:
Indeed. It's more preferable.